src/main/java/io/confluent/connect/hdfs/TopicPartitionWriter.java

```diff
@@ -482,13 +482,8 @@ public void buffer(SinkRecord sinkRecord) {
     buffer.add(sinkRecord);
   }
 
-  /**
-   * Get the offset that was last committed to HDFS, or null if nothing was committed yet.
-   *
-   * @return the last committed offset; may be null
-   */
-  public Long committedOffset() {
-    return committedOffset;
+  public long offset() {
```

Comment:
@cyrusv, I'm not sure I understand why we're returning 1 past what was last committed to HDFS. We need to verify that what is committed to the consumer offsets exactly matches what is committed to HDFS. I'm not convinced this really identified/fixed the bug.

Second, `offset` is initialized to `-1L`, so is it ever possible that this method return that? Before this change, the offset getter might return null, but the code in DataWriter looked for this case and never wrote that into the offsets.

Finally, we lost JavaDoc - can we put it back to explain what offset this actually returns?

Reply:
So I wonder if this is the source of the confusion: the consumer offset should point to the offset of the *next record to be consumed, whereas the prior change recorded for the consumer offset the *last record committed to HDFS*.

It'd be good to document this, though I still think the `offset()` method should mention why it's 1 past the last committed offset and why (to match the consumer offsets). Also, we still need to handle the -1 case mentioned above.

Reply:
Thanks for the feedback, @rhauch  -- I've renamed some variables and added some documentation to clarify the work done here. You can see the work is most closely related to `284fbc2`, for reference.

Reply:
I decided to stick to a primitive with `-1` initial value, since a lot of logic in the `TopicPartitionWriter` tests against `-1` and I think keeping this code as-is (relative to pre-`284fbc2`) with a clearer name will provide the most stability. I could be convinced that `null` initial value would be preferable if we'd rather have NPEs than misleading long values, even though I decided it wasn't worth the tradeoff since we've seen this structure work successfully in many production envs.